src/hvm/check_net_size.rs

```diff
@@ -2,16 +2,22 @@ use super::tree_children;
 use crate::{diagnostics::Diagnostics, fun::Name};
 use hvm::ast::{Book, Net, Tree};
 
-pub const MAX_NET_SIZE: usize = 64;
+pub const MAX_NET_SIZE_C: usize = 4095;
+pub const MAX_NET_SIZE_CUDA: usize = 64;
+
+pub fn check_net_sizes(book: &Book, diagnostics: &mut Diagnostics, cmd: &str) -> Result<(), Diagnostics> {
```

Comment:
The CLI should be decoupled from Bend as a library, so we don't want to pass the command as a string.
Instead this value should be part of the compiler options struct, and set by the CLI in `compile_opts_from_cli`